### PR TITLE
Fixing XSS Vulnerabilities:STS Component

### PR DESCRIPTION
**Pull request by kasungayan**

### CODE REVIEW

File: components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/generic-sts/remove-trusted-service.jsp [hunk 1]
```diff
@@ -55,7 +56,7 @@
 boolean qpplicationComponentFound = CarbonUIUtil.isContextRegistered(config, "/application/");
 if (qpplicationComponentFound) {
 %>
-    location.href = '../application/configure-service-provider.jsp?spName=<%=spName%>&action=<%=action%>&serviceName=<%=address%>';
+    location.href = '../application/configure-service-provider.jsp?spName=<%=Encode.forJavaScriptBlock(spName)%>&action=<%=Encode.forJavaScriptBlock(action)%>&serviceName=<%=Encode.forJavaScriptBlock(address)%>';
```
Comment (malithie) [inline components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/generic-sts/remove-trusted-service.jsp]: each entry encoded here should be encoded for uri component

File: components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/generic-sts/sts.jsp [hunk 2]
```diff
@@ -138,7 +139,7 @@
             	if (qpplicationComponentFound) {
 %> 
 <script>
-   location.href = '../application/configure-service-provider.jsp?action=update&display=serviceName&spName=<%=spName%>&serviceName=<%=address%>';
+   location.href = '../application/configure-service-provider.jsp?action=update&display=serviceName&spName=<%=Encode.forJavaScriptBlock(spName)%>&serviceName=<%=Encode.forJavaScriptBlock(address)%>';
```
Comment (malithie) [inline components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/generic-sts/sts.jsp]: each entry encoded here should be encoded for uri component

File: components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/generic-sts/sts.jsp [hunk 3]
```diff
@@ -186,9 +187,9 @@
                 </td>
                 <td>
                     <div style="height:30px;">
-                        <a href="javascript:document.location.href='<%=serverUrl+"?wsdl"%>'"
+                        <a href="javascript:document.location.href='<%=Encode.forHtmlAttribute(serverUrl)+"?wsdl"%>'"
```
Comment (malithie) [inline components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/generic-sts/sts.jsp]: Here javascript execution context is being invoked. Thus javascript encoding should be performed. However, serverurl seems to be constructed internally and trusted. So it's not necessary to apply encoding here

File: components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/identity-sts/configure.jsp [hunk 4]
```diff
@@ -118,11 +119,11 @@ try {
 				<table class="normal" cellspacing="0">
 		          <tr>
 	                      <td><fmt:message key='card.name'/><font class="required">*</font> </td>
-	                      <td> <input type="text" name="cardName" id="cardName" value="<%=cardIssuer.getCardName()%>" /></td>	                     
+	                      <td> <input type="text" name="cardName" id="cardName" value="<%=Encode.forHtmlAttribute(cardIssuer.getCardName())%>" /></td>
 	                  </tr>
 	                  <tr>
 	                      <td><fmt:message key='valid.period'/><font class="required">*</font> </td>  
-	                      <td><input type="text" name="validPeriod" id="validPeriod" value="<%=cardIssuer.getValidPeriodInDays()%>" /></td>                  
+	                      <td><input type="text" name="validPeriod" id="validPeriod" value="<%=cardIssuer.getValidPeriodInDays()%>" /></td>
```
Comment (malithie) [inline components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/identity-sts/configure.jsp]: encode for html attribute

File: components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/identity-sts/configure.jsp [hunk 5]
```diff
@@ -131,13 +132,13 @@ try {
...
+	                          <input type="hidden" name='<%=Encode.forHtmlAttribute(cardIssuer.getSupportedTokenTypes()[i].getTokenType()) + "hidden"%>' id='<%=Encode.forHtmlAttribute(cardIssuer.getSupportedTokenTypes()[i].getTokenType()) + "hidden"%>' value="true"/>
+	                          <input type='checkbox' name='<%=Encode.forHtmlAttribute(cardIssuer.getSupportedTokenTypes()[i].getTokenType())%>' id='<%=Encode.forHtmlAttribute(cardIssuer.getSupportedTokenTypes()[i].getTokenType())%>' checked='checked' onclick="setValue('<%=Encode.forJavaScriptAttribute(cardIssuer.getSupportedTokenTypes()[i].getTokenType())%>','<%=Encode.forHtmlAttribute(cardIssuer.getSupportedTokenTypes()[i].getTokenType())+"hidden"%>')" /> <%=Encode.forHtmlContent(cardIssuer.getSupportedTokenTypes()[i].getTokenType())%> &nbsp;
```
Comment (malithie) [inline components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/identity-sts/configure.jsp]: Each parameter passed to the javascript function should be encoded for javascript attribute here

File: components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/identity-sts/index.jsp [hunk 6]
```diff
@@ -82,15 +83,15 @@ try {
 		              </tr>
 		              <tr>
 	                      <td><fmt:message key='card.name'/></td>
-	                      <td><%=cardIssuer.getCardName()%></td>	                     
+	                      <td><%=Encode.forHtmlContent(cardIssuer.getCardName())%></td>
 	                  </tr>
 	                  <tr>
 	                      <td><fmt:message key='valid.period'/></td>
-	                      <td><%=cardIssuer.getValidPeriodInDays()%></td>	                     
+	                      <td><%=cardIssuer.getValidPeriodInDays()%></td>
```
Comment (malithie) [inline components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/identity-sts/index.jsp]: encode for html content

Comment (kasungayan) [inline components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/identity-sts/index.jsp]: cardIssuer.getValidPeriodInDays() returns an integer, hence doesn't need to Encode.

File: components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/identity-sts/configure.jsp (hunk 4 above)
Comment (kasungayan) [inline components/sts/org.wso2.carbon.identity.sts.mgt.ui/src/main/resources/web/identity-sts/configure.jsp]: cardIssuer.getValidPeriodInDays() returns an integer, hence doesn't need to Encode.

**Comment by malithie** [comment]
Can merge

**Pull request merged**